### DISCUSSION TITLE
C&C Blocks: Suggestions around features to satisfy Legal Requirements

### DISCUSSION THREAD

**Discussion by danielxescobar** [woocommerce/woocommerce#45069] [WooCommerce, Blocks, and Site Editing]
Hello,

I love the interface of the checkout. It's a huge improvement to the previous checkout, however, I will be opting out from using it at the moment as it does not have a checkbox where it requires the individual to actively select the terms and conditions and it does not state something along the lines of, "Your personal data will be used to process your order, support your experience throughout this website, and for other purposes described in our privacy policy." with links to both of those pages.

A couple other suggestions I would like to see implemented are:

1. To have the "Create an account?" check box be moved all the way to the bottom after entering their payment information, but be rephrased as "Create an account and save payment information for future use?". This way it gives them a bit of an incentive to create an account and there's less resistance since it's at the end of checkout after they've already entered all of their info. Plus, both questions are consolidated into one check box, so less steps for them to take.
2. To add a checkbox underneath the email field that we can elect to turn on and off if it will be pre-selected with custom wording we can enter such as, "Subscribe for exclusive offers, builds, updates, and more!"

**Comment by mikejolley** [comment]
Hello,

The addition of a Terms and Conditions checkbox is something we're working on currently, so it will be part of a future version. The ability to add custom text such as _"Your personal data will be used to process your order, support your experience throughout this website, and for other purposes described in our privacy policy."_  will also be possible. No ETA for this to be released but I'd expect it sometime this year.

Regarding the other suggestions, the subscribe checkbox would be part of your newsletter plugin - we're also working on ways to allow such plugins to add checkboxes for this purpose.

R/e the create an account box, @vivialice is this something we should consider as part of Checkout i2? Asking because we currently have it has a toggle option of the contact information step.

Thanks

**Comment by vivialice** [comment]
@mikejolley Yes, I think we can look at this. I think the above suggestion could work, but I'd also recommend having this step as part of the confirmation page to keep the customer focused on completing the purchase. With a prompt of something like: "Thanks for your purchase! Create an account to keep track of your order and faster checkout next time you shop with us". Would this scenario be possible to implement?

**Comment by mikejolley** [comment]
@vivialice Right now, not possible because the confirmation step is not part of the Checkout Block itself (we're using the core "thanks" page). If we included a confirmation step in the Block we could, but that's probably a future roadmap item (cc @nerrad).

**Comment by nerrad** [comment]
Could the account checkbox be something we convert to a block at some point in the future which would allow for some customization around the text? Or is it something we would need to do as a block _now_? My question is mostly around avoiding the potential impact of migration to avoid invalid block issues for existing instances of the checkout block if/when it happens.

Regarding the confirmation step or even potentially moving the account creation to the order received/thanks page - I agree this is a future roadmap item.

**Comment by nerrad** [comment]
> Regarding the confirmation step or even potentially moving the account creation to the order received/thanks page - I agree this is a future roadmap item.

I've created #4697 to track this. I think we can go ahead and close this issue since the one request will be satisfied by the release of Checkout I2 and the create account behaviour/functionality is addressed as a part of the issue I just created.
